### DISCUSSION TITLE
SO stopped to work

### DISCUSSION THREAD

**Discussion by ben-sec** [Security-Onion-Solutions/securityonion#6684] [Unsupported Versions] [answered]
Hello!

My distributed SO 2.3.90 (incl. latest hotfixes) deployment stopped suddenly to work last night (no more data in ES so-* indices since about 1:00 am, however a custom index still stores data successfully). so-status showed no errors (manager, search and forward nodes are all OK), but I found this in Grafana for my search node:

![image](https://user-images.githubusercontent.com/81571847/147207828-91d77bf2-d4e5-4848-b6aa-8c01ee469a2f.png)

What happened at 1:00 am? I rebooted the serach node and the manager without success. After the reboot elastalert did not start  and crashed  after a few minutes after I started it manually, Any ideas what the screenshot above might indicate or where I should start to investigate? Do you recommend an update to 2.3.91 in the current situation?

Thanks in advance!

Cheers, Ben

**Comment by ben-sec** [comment]
OK, fixed the elastalert error, but this was not related to the overall issue I have.

**Comment by ben-sec** [comment]
Couldnt't wait and installed SO 2.3.91: Status OK on all nodes and containers, but still noch data in ES so-* indices. Merry Christmas!

**Comment by ben-sec** [comment]
You got me wrong, SO is **still not working**, even after the update to 2.3.91. Your help is still highly appreciated!

Cheers Ben

**Comment by dougburks** [comment, marked as answer]
Have you checked the logs in `/opt/so/log/` for additional clues?

**Reply by ben-sec** [reply]
I just checked SOC again and found the following status for my searche node:

![image](https://user-images.githubusercontent.com/81571847/147573812-32844d45-d1d9-45ce-acd2-8edc46072ca8.png)

And now I see that the root partition ran full since SO stopped to work on Dec 23:

![image](https://user-images.githubusercontent.com/81571847/147574685-9ff2752a-71ca-42e3-89f6-cb8170760434.png)

I found the following Elasticsearch error on my Manager:

`[2021-12-23T07:12:20,759][WARN ][org.elasticsearch.transport.SniffConnectionStrategy] fetching nodes from external cluster [my_searchnode] failed`

The Elasticsearch log of my Search node is full with millions of lines like:

`[2021-12-23T23:58:38,971][CRITICAL][org.elasticsearch.deprecation.cluster.metadata.MetadataCreateIndexService] index [so-zeek-2021.12.23] matches multiple legacy templates [so-common, so-zeek], composable templates will only matc        h a single template`

What's going on here and how can I get my search node under control again?

**Reply by dougburks** [reply]
> And now I see that the root partition ran full since SO stopped to work on Dec 23:

Is that your manager or your search node?  How is it partitioned?

Take a look at the filesystem to see what is using the majority of the space. Clear old logs if you can to get disk usage back to a normal level.

> [2021-12-23T07:12:20,759][WARN ][org.elasticsearch.transport.SniffConnectionStrategy] fetching nodes from external cluster [my_searchnode] failed

Seems like your manager is having trouble connecting to your search node. Could be related to the above but go ahead and double-check networking and firewall rules just in case.

**Reply by ben-sec** [reply]
That's the search node filled up with huge Elasticssearch and Logstash error logs since December 23. The errors above are all from December 23 when disk space was not yet an issue. The Logstash log on the search node shows on December 23 the following error (millions of them):

`[2021-12-23T00:00:01,687][WARN ][logstash.outputs.elasticsearch] Could not index event to Elasticsearch. {:status=>400 ... "error"=>{"type"=>"val        idation_exception", "reason"=>"Validation Failed: 1: this action would add [1] shards, but this cluster currently has [1000]/[1000] maximum normal shards open;`

It's still unclear to me what happened suddenly, since we didn't change a thing ... even more unclear how to get things fixed.

**Reply by dougburks** [reply]
Check your Curator `close` settings:
https://docs.securityonion.net/en/2.3/elasticsearch.html#closing-indices

and `log_size_limit` setting:
https://docs.securityonion.net/en/2.3/elasticsearch.html#deleting-indices

**Comment by ben-sec** [comment]
BTW: How can I stop SO containers permanently? After `so-*-stop` or `docker stop id` SO will relaunch the containers after a few minutes. I have to stop Logstash and ES to stop the error log nightmare!

**Reply by m0duspwnens** [reply]
If you stop the containers, next time Salt runs a highstate, it will restart them. To stop this from happening:

All run as root:

1. Stop the salt-minion service: `systemctl stop salt-minion`

2. Make sure all salt-minion processes are stopped: `pkill -9 -f salt-minion`

3. Remove cronjob that check salt-minion health and restarts if failed: `crontab -e`
Remove lines from cron: 
```
# SALT_CRON_IDENTIFIER:so-salt-minion-check
*/5 * * * * /usr/sbin/so-salt-minion-check -q
```

4. Stop the containers you want

5. When you want to return to normal, run `salt-call state.highstate` and the salt-minion as well as stopped containers will start. The cronjob will also be put back in place.

**Reply by ben-sec** [reply]
Thanks!

**Comment by ben-sec** [comment]
Fixed it. It was the Curator close setting. I set some of them in the past to [200 days](https://github.com/Security-Onion-Solutions/securityonion/discussions/5556) and that killed ES on Dec 23. I reduced them to 100 and now data is comming in again. Did that lead to this limit of 1000 Shrads that I reached? Can I increase the limit of Shrads or what do I have to do to increase the log retention (I reached about 150 days on December 23, but did not exceed the log_size_limit or disk space)?

I will pick this up again after my vacation on Jan 3 2022. Thanks for your support!

Cheers, Ben

**Reply by dougburks** [reply]
https://docs.securityonion.net/en/2.3/elasticsearch.html#shards
https://docs.securityonion.net/en/2.3/elasticsearch.html#closing-indices
